server/servers/api/src/test/scala/com/prisma/api/filters/ExtendedRelationFilterSpec.scala

```diff
@@ -0,0 +1,429 @@
+package com.prisma.api.filters
+
+import com.prisma.{IgnoreMySql, IgnorePostgres}
+import com.prisma.api.ApiSpecBase
+import com.prisma.shared.schema_dsl.SchemaDsl
+import org.scalatest._
+
+class ExtendedRelationFilterSpec extends FlatSpec with Matchers with ApiSpecBase {
+
+  override def runSuiteOnlyForActiveConnectors = true
+
+  val project = SchemaDsl.fromString() { """type Artist {
+                                         |  id: ID! @unique
+                                         |  ArtistId: Int! @unique
+                                         |  Name: String!
+                                         |  Albums: [Album!]!
+                                         |}
+                                         |
+                                         |type Album {
+                                         |  id: ID! @unique
+                                         |  AlbumId: Int! @unique
+                                         |  Title: String!
+                                         |  Artist: Artist!
+                                         |  Tracks: [Track!]!
+                                         |}
+                                         |
+                                         |type Genre {
+                                         |  id: ID! @unique
+                                         |  GenreId: Int! @unique
+                                         |  Name: String!
+                                         |  Tracks: [Track!]!
+                                         |}
+                                         |
+                                         |type MediaType {
+                                         |  id: ID! @unique
+                                         |  MediaTypeId: Int! @unique
+                                         |  Name: String!
+                                         |  Tracks: [Track!]!
+                                         |}
+                                         |
+                                         |type Track {
+                                         |  id: ID! @unique
+                                         |  TrackId: Int! @unique
+                                         |  Name: String!
+                                         |  Album: Album!
+                                         |  MediaType: MediaType!
+                                         |  Genre: Genre!
+                                         |  Composer: String
+                                         |  Milliseconds: Int!
+                                         |  Bytes: Int!
+                                         |  UnitPrice: Float!
+                                         |}
+                                         |""" }
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    database.setup(project)
+  }
+
+  override def beforeEach() = {
+    super.beforeEach()
```

Comment:
Since you are only querying data, this could also go into `beforeAll`, right?

Reply:
done